pkg/nrop/nrop_scheduler.go

```diff
@@ -0,0 +1,277 @@
+	errorMsg string
+	// api client to interact with the cluster.
+	apiClient goclient.Client
+}
+
+// NewSchedulerBuilder creates a new instance of NUMAResourcesScheduler.
+func NewSchedulerBuilder(
+	apiClient *clients.Settings, name, nsname string) *SchedulerBuilder {
+	glog.V(100).Infof(
+		"Initializing new NUMAResourcesScheduler structure with the following name: %s in namespace %s",
+		name, nsname)
+
+	if apiClient == nil {
+		glog.V(100).Infof("NUMAResourcesScheduler 'apiClient' cannot be empty")
+
+		return nil
+	}
+
+	builder := &SchedulerBuilder{
+		apiClient: apiClient.Client,
+		Definition: &nropv1.NUMAResourcesScheduler{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the NUMAResourcesScheduler is empty")
+
+		builder.errorMsg = "NUMAResourcesScheduler 'name' cannot be empty"
+
+		return builder
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The nsname of the NUMAResourcesScheduler is empty")
+
+		builder.errorMsg = "NUMAResourcesScheduler 'nsname' cannot be empty"
+
+		return builder
+	}
+
+	return builder
+}
+
+// PullScheduler pulls existing NUMAResourcesScheduler from cluster.
+func PullScheduler(apiClient *clients.Settings, name, nsname string) (*SchedulerBuilder, error) {
+	glog.V(100).Infof("Pulling existing NUMAResourcesScheduler %s in namespace %s from the cluster",
+		name, nsname)
+
+	if apiClient == nil {
+		glog.V(100).Infof("The apiClient is empty")
+
+		return nil, fmt.Errorf("NUMAResourcesScheduler 'apiClient' cannot be empty")
+	}
+
+	builder := SchedulerBuilder{
+		apiClient: apiClient.Client,
+		Definition: &nropv1.NUMAResourcesScheduler{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the NUMAResourcesScheduler is empty")
+
+		return nil, fmt.Errorf("NUMAResourcesScheduler 'name' cannot be empty")
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The nsname of the NUMAResourcesScheduler is empty")
+
+		return nil, fmt.Errorf("NUMAResourcesScheduler 'nsname' cannot be empty")
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("NUMAResourcesScheduler object %s does not exist", name)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Get fetches the defined NUMAResourcesScheduler from the cluster.
+func (builder *SchedulerBuilder) Get() (*nropv1.NUMAResourcesScheduler, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting NUMAResourcesScheduler %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	nrosObj := &nropv1.NUMAResourcesScheduler{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name:      builder.Definition.Name,
+		Namespace: builder.Definition.Namespace,
+	}, nrosObj)
+
+	if err != nil {
+		return nil, err
```

Comment:
Please add log message:
			"NUMAResourcesScheduler object %s does not exist in namespace %s",
			builder.Definition.Name, builder.Definition.Namespace)